### PR TITLE
Ensure that guides have consistent formatting

### LINKED ISSUE #617: Guides have inconsistent formatting
The guides in this repository do not have consistent formatting or naming.

Specifically:
 - All guides should be using [GitHub Flavored Markdown](https://github.github.com/gfm/)
 - Heading levels should be incremental
 - Links should be defined at the bottom of each section, rather than inline

### PR DESCRIPTION
**Pull request by WilHall**
Fixes #617.

This change ensures that all the guides have consistent Markdown
formatting. It does not make any changes to the content.

Formatting changes include:

 - Use GitHub style markdown
 - Make heading levels consistent and incremental
 - Make external links extracted instead of inline
 - Normalized line length to 80 columns

Additionally, I came across and fixed some editorial issues while
reviewing all the guides:

 - Fixed/removed broken links
 - Fixed capitalization of technologies
 - Fixed spelling errors
 - Improved the link text for some external links
 - Replaced references to "Slack" with "Basecamp"

### CODE REVIEW

**Comment by jferris** [comment]
Thanks for putting this together - consistent formatting certainly makes sense for a document about consistent formatting.

It looks like some of the lines were reformatted to use different line lengths. We've traditionally stuck to 80 character limits for lines, and changing the length here makes it harder to tell what actually changed in some places.

File: CODE_OF_CONDUCT.md [hunk 1]
```diff
@@ -1,6 +1,5 @@
 # Code of Conduct
 
-By participating in this project, you agree to abide by the
-[thoughtbot code of conduct][tb-coc].
+By participating in this project, you agree to abide by the [thoughtbot code of conduct].
 
-[tb-coc]: https://thoughtbot.com/open-source-code-of-conduct
+[thoughtbot code of conduct]: https://thoughtbot.com/open-source-code-of-conduct
```
Comment (ericwbailey) [inline CODE_OF_CONDUCT.md:5]: I've been treating Code of Conduct as a proper noun. Do you think that's worth keeping, or is it distracting?

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -1,28 +1,25 @@
 # Contributing
 
-We love contributions from everyone. By participating in this project, you
-agree to abide by the [thoughtbot code of conduct][tb-coc].
+We love contributions from everyone. By participating in this project, you agree to abide by the [thoughtbot code of
```
Comment (ericwbailey) [inline CONTRIBUTING.md:3]: Echoing [Joe's comment](https://github.com/thoughtbot/guides/pull/618#issuecomment-654188058) about 80 char line breaks. I learned that if you use VS Code there's a command palette option to auto-format this!

File: README.md [hunk 3]
```diff
@@ -58,7 +58,7 @@ Guides for working together, getting things done, programming well, and programm
 - [Objective-C](/objective-c/)
 - [Python](/python/)
 - [Ruby](/ruby/)
-- [SASS](/sass/)
+- [Sass](/sass/)
```
Comment (ericwbailey) [inline README.md:61]: 🙏

File: code-review/README.md [hunk 4]
```diff
@@ -1,75 +1,65 @@
...
-  branch. Do not squash until the branch is ready to merge. Reviewers should be
-  able to read individual updates based on their earlier feedback.
-* Seek to understand the reviewer's perspective.
-* Try to respond to every comment.
-* Wait to merge the branch until continuous integration (TDDium, Travis CI,
-  CircleCI, etc.) tells you the test suite is green in the branch.
-* Merge once you feel confident in the code and its impact on the project.
-* Final editorial control rests with the pull request author.
+- Be grateful for the reviewer's suggestions. ("Good call. I'll make that change.")
+- A common axiom is "Don't take it personally. The review is of the code, not you." We used to include this, but now
```
Comment (ericwbailey) [inline code-review/README.md:27]: Potentially out of scope, but I don't like this point. Maybe thread in [a link to Edward's blog post](https://thoughtbot.com/blog/empathy-online) to communicate that the reviewer also has a responsibility to communicate well?

File: data/README.md [hunk 5]
```diff
@@ -4,109 +4,93 @@ A guide for managing a series of tubes.
...
+  that queue out of order.
+- When partitioning messages in a single topic, ensure that it's acceptable for messages from different partitions to be
+  processed out of order.
 
 ## Glossary
 
-*Data Engineering*: the task of building infrastructure and process for
-ingesting, processing, and aggregating data so that it can be displayed to users
-or made available to data scientists.
+_Data Engineering_: the task of building infrastructure and process for ingesting, processing, and aggregating data so
```
Comment (ericwbailey) [inline data/README.md:27]: Wonder if we should make these `<dl>` lists.

File: git/README.md [hunk 6]
```diff
@@ -24,48 +24,60 @@ A guide for programming within version control.
 
 Create a local feature branch based off master.
```
Comment (ericwbailey) [inline git/README.md:25]: Thoughts on `main` instead of `master`?

File: git/README.md [hunk 7]
```diff
@@ -24,48 +24,60 @@ A guide for programming within version control.
...
 
-If you've created more than one commit,
-[use `git rebase` interactively](https://help.github.com/articles/about-git-rebase/) to squash them into cohesive
-commits with good messages:
+If you've created more than one commit, [use `git rebase` interactively] to squash them into cohesive commits with good
+messages:
 
-    git rebase -i origin/master
+```console
+git rebase -i origin/master
```
Comment (ericwbailey) [inline git/README.md:66]: `git rebase -i origin/main`, if the previous comment is accepted.

File: git/README.md [hunk 8]
```diff
@@ -91,28 +105,38 @@ When satisfied, they comment on the pull request `Ready to merge.`
 Rebase interactively. Squash commits like "Fix whitespace" into one or a small number of valuable commit(s). Edit commit
 messages to reveal intent. Run tests.
 
-    git fetch origin
-    git rebase -i origin/master
+```console
+git fetch origin
+git rebase -i origin/master
```
Comment (ericwbailey) [inline git/README.md:110]: `main` here, as well. Probably best to do a find/replace.

File: sass/README.md [hunk 9]
```diff
@@ -101,7 +100,7 @@ Comprehensive example of ordering items within a declaration block:
   }
 
   &::before {
-    content: "";
+    content: '';
```
Comment (ericwbailey) [inline sass/README.md:103]: I think house style is double quotes for `content`, as it's still a CSS property.

File: security/README.md [hunk 10]
```diff
@@ -4,83 +4,75 @@ A guide for practicing safe web.
...
 
-* Lock your device when you are away from it.
-* Don't leave your devices unattended in an unsecured area.
-* Install a device tracking and remote data wipe tool such as [Prey].
+- Lock your device when you are away from it.
+- Don't leave your devices unattended in an unsecured area.
+- Install a device tracking and remote data wipe tool such as [Prey].
 
-[Prey]: https://www.preyproject.com/
+[prey]: https://www.preyproject.com/
```
Comment (ericwbailey) [inline security/README.md:49]: Do we still use Prey?

File: CODE_OF_CONDUCT.md (hunk 1 above)
Comment (mike-burns) [reply CODE_OF_CONDUCT.md:5]: +1. I treat our Handbook and Guides as a proper noun, and I think by analogy I'd also treat our Code of Conduct as a proper noun. (But I wouldn't treat the idea of a code of conduct as a proper noun.)

File: security/README.md (hunk 10 above)
Comment (mike-burns) [reply security/README.md:49]: TBH, I haven't been.

File: CONTRIBUTING.md (hunk 2 above)
Comment (WilHall) [reply CONTRIBUTING.md]: I use `vim`, and I have an autoformatter that _should_ do this. It turns out that at some point, Prettier started fighting with it which was causing some pretty inconsistent behavior 😬 

I have normalized them to 80 characters now 🙂

Comment (ericwbailey) [reply CONTRIBUTING.md]: https://i.giphy.com/media/qN7NZR3Q5R2mY/giphy.webp

File: data/README.md (hunk 5 above)
Comment (WilHall) [reply data/README.md]: ❤️ https://github.com/thoughtbot/guides/issues/619

File: git/README.md (hunk 8 above)
Comment (WilHall) [reply git/README.md:112]: I did a pass across all the guides 😉

File: sass/README.md (hunk 9 above)
Comment (WilHall) [reply sass/README.md:113]: _prettier_

File: security/README.md (hunk 10 above)
Comment (WilHall) [reply security/README.md:55]: I do but I don't like it

**Comment by WilHall** [comment]
@jferris 
> It looks like some of the lines were reformatted to use different line lengths. We've traditionally stuck to 80 character limits for lines

I've updated this, along with my autoformatter 🙂

File: code-review/README.md (hunk 4 above)
Comment (WilHall) [reply code-review/README.md]: I greatly simplified this and linked to that article instead of the external resource. Thoughts?

**Pull request merged**